packages/vscode-extension/src/commands/CompareFile.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { UNRESTRICTED_FOLDERS } from '../ApplicationConstants';
+import { COMPARE_FILE, IMPORT_FILES } from '../service/TranslationKeys';
+import { actionResultStatus, ApplicationConstants } from '../util/ExtensionUtil';
+import FileImportCommon from './FileImportCommon';
+
+export default class CompareFile extends FileImportCommon {
+	private static readonly COMMAND_NAME = 'comparefile';
+    private static readonly TEMP_FOLDER_PREFIX = 'suitecloud-vscode-extension-compare-file-';
+
+	constructor() {
+		super(CompareFile.COMMAND_NAME);
+	}
+
+	protected validateBeforeExecute() {
+		const superValidation = super.validateBeforeExecute();
+		if (!superValidation.valid) {
+			return superValidation;
+		}
+
+		if (!this.activeFileIsUnderUnrestrictedFolder()) {
+			return this.unsuccessfulValidation(this.translationService.getMessage(IMPORT_FILES.ERROR.NOT_ALLOWED_FOLDER));
+		}
+
+		return this.successfulValidation();
+	}
+
+	protected async execute() {
+		const activeFilePath = this.activeFile!;
+		const tempFolderPath = fs.mkdtempSync(path.join(os.tmpdir(), CompareFile.TEMP_FOLDER_PREFIX));
```

Comment:
I'm wondering if the temp files are cleaning ok... in [this page](https://stackoverflow.com/questions/43607942/cleaning-temporary-files-after-file-upload-with-restify) it implies that in Unix it could not be cleaned until system is reboot.
I would say to put a last step (if possible) for cleaning these folder with fs.unlink()

Also, while searching about this, I found this[ other page](https://advancedweb.hu/secure-tempfiles-in-nodejs-without-dependencies/) explain also some possible problems with macOS and os.tmpdir() method but it may not apply to us, just sending it to you in case you find it useful.

Reply:
Unfortunately, we can't do cleanup since the diff window still needs to access the file. If we do cleanup, the diff window will not be able to show the difference.

Reply:
Ok, and what about detecting when the diff window is closed and then removing the file? Maybe it is not possible, so in that case forget my comment.

Reply:
I think we could leave it there. no?

Reply:
I don't think it's a big deal if we leave some small file in the temp folder :)

Reply:
There's no mechanism to capture the closing of the window.